cpp/src/trajectory/subset_trajectories.cu

```diff
@@ -45,72 +46,67 @@ struct subset_functor {
     }
 
     template <typename T, std::enable_if_t< is_supported<T>() >* = nullptr>
-    uint32_t operator()(const gdf_column& id,
-                        const gdf_column& in_x, const gdf_column& in_y,
-                        const gdf_column& in_id, const gdf_column& in_timestamp,
-                        gdf_column& out_x, gdf_column& out_y,
-                        gdf_column& out_id, gdf_column& out_timestamp)
+    gdf_size_type operator()(const gdf_column& id,
+                             const gdf_column& in_x, const gdf_column& in_y,
+                             const gdf_column& in_id,
+                             const gdf_column& in_timestamp,
+                             gdf_column& out_x, gdf_column& out_y,
+                             gdf_column& out_id, gdf_column& out_timestamp)
     {
-        T* in_x_ptr = static_cast<T*>(in_x.data);
-        T* in_y_ptr = static_cast<T*>(in_y.data);
-        uint32_t* in_id_ptr = static_cast<uint32_t*>(in_id.data);
-        cuspatial::its_timestamp* in_ts_ptr =
-            static_cast<cuspatial::its_timestamp*>(in_timestamp.data);
-        uint32_t* id_ptr = static_cast<uint32_t*>(id.data);
-
-        cudaStream_t stream{0};
-        auto exec_policy = rmm::exec_policy(stream)->on(stream);
-
+        gdf_size_type num_hit{0};
         gdf_size_type num_id{id.size};
         gdf_size_type num_rec{in_id.size};
 
-        rmm::device_vector<uint32_t> temp_id(id_ptr, id_ptr + num_id);     
-        thrust::sort(exec_policy, temp_id.begin(), temp_id.end());            
-        thrust::device_vector<bool> hit_vec(num_rec);
-        thrust::binary_search(exec_policy, temp_id.cbegin(), temp_id.cend(),
-                              in_id_ptr, in_id_ptr + num_rec, hit_vec.begin());
-
-
-        uint32_t num_hit = thrust::count(exec_policy, hit_vec.begin(),
-                                            hit_vec.end(), true);
+        if (num_id > 0 && id.data != nullptr && num_rec > 0) {
+            int32_t* in_id_ptr = static_cast<int32_t*>(in_id.data);
```

Comment:
The `CUDF_EXPECTS` for `in_id.dtype` allows it to be int8/16/32/64, but this is always casting to `int32_t`.

Something should change here, and I'm guessing it's making the `CUDF_EXPECTS` more restrictive.

Reply:
Fixed.